src/herbivoresAndCarnivores.js

```diff
@@ -2,14 +2,38 @@
 
 class Animal {
   // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
   // write your code here
+  constructor(name) {
+    super(name);
+
+    this.hidden = false;
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
   // write your code here
+  bite(victim) {
+    if (victim instanceof Carnivore || victim.hidden) {
+      return;
+    }
+
+    victim.health -= 50;
```

Comment:
```suggestion
    if (victim instanceof Herbivore && !victim.hidden) {
       victim.health -= 50;
    }
```
We can improve this conditions a bit